File: integration-tests/mongodb-rest-data-panache/src/main/java/io/quarkus/it/mongodb/rest/data/panache/BookRepository.java
package io.quarkus.it.mongodb.rest.data.panache;

import jakarta.enterprise.context.ApplicationScoped;

import io.quarkus.mongodb.panache.PanacheMongoRepositoryBase;

@ApplicationScoped
public class BookRepository implements PanacheMongoRepositoryBase<Book, String> {
}

File: integration-tests/mongodb-panache-kotlin/src/main/kotlin/io/quarkus/it/mongodb/panache/book/BookRepository.kt
package io.quarkus.it.mongodb.panache.book

import io.quarkus.mongodb.panache.kotlin.PanacheMongoRepository
import jakarta.enterprise.context.ApplicationScoped

@ApplicationScoped
class BookRepository(private val dummyService: DummyService) : PanacheMongoRepository<Book>
